frontend/transaction/transaction.js

```diff
@@ -538,57 +584,148 @@ function popupWithdraw(dataItem) {
+        <input type="number" class="form-control value" id="InputCost${id}" aria-label="Cost" placeholder="Cost">
+        <input type="number" class="form-control value" id="InputAmount${id}" placeholder="Amount"> 
     `;
 
     return liItem;
 }
 
-function appendAddBtn() {
-    let liAddBtn = document.createElement("li");
-    liAddBtn.setAttribute("class", "info")
-    liAddBtn.innerHTML = `
-        <div class="withdraw-form">
-            <button class="btn btn-primary" id="addRow" type="submit">+</button>
-        </div>
-    `;
-    return liAddBtn;
+
+// ===============+ API calls ====================
+
+function convertDepositToJSON(element) {
+    var output = [];
+    let ul = element.childNodes
+    
+    for (var i = 0; i < ul.length-1; i++) {
+        let filteredInput = jQuery(ul[i].childNodes).filter('*') // Removes text nodes
+        console.log("I is filtered", filteredInput)
+        output.push({
+            "desc": filteredInput[1].value,
+            "qty": filteredInput[2].value,
+        });
+    }
+    return output;    
+}
+
+function convertWithdrawalsToJSON(element) {
+    var output = [];
+    let ul = element.childNodes
+    
+    for (var i = 0; i < ul.length-1; i++) {
+        let filteredInput = jQuery(ul[i].childNodes).filter('*') // Removes text nodes
+        output.push({
+            "type": filteredInput[0].value,
+            "cost": filteredInput[1].value,
+            "amount": filteredInput[2].value,
+        });
+    }
+    return output;    
+}
+
+function submitDepositList() {
+    let root = document.getElementById("InputItemsList");
+    let depositOutput = convertDepositToJSON(root)
+    console.log(depositOutput);
+
+    // FIXME: Implement API route here
+    // NOTE: need to build the json payload
+    // {
+    //     "id": 1,
+    //     "orderInfo": [
+    //         {
+    //             "desc": 'Flower bookmark', 
+    //             "qty": '3
+    //         },
+    //         {
+    //             "desc": 'Enchanted bookmark', 
+    //             "qty": '5'
+    //         },
+    //     ]
+    // }
+
+    let data = {
+        "id": currentDepositData.id,
+        "orderInfo": depositOutput
+    }
+
+    closePopup()
 }
 
+function submitWithdrawList() {
+    let root = document.getElementById("InputItemsList");
+    let withdrawalOutput = convertWithdrawalToJSON(root)
+    console.log(withdrawalOutput);
+
+    // FIXME: Implement API route here
+    // NOTE: need to build the json payload
+    // {
+    //     "id": 1,
+    //     "orderInfo": [
+    //         {
+    //             "desc": 'Black yarn', 
+    //             "qty": '2'
+    //             "cost": '24.55'
+    //         },
+    //         {
+    //             "desc": 'Pink yarn', 
+    //             "qty": '4'
+    //             "cost": '32.44'
+    //         },
+    //     ]
+    // }
+
+    let data = {
+        "id": currentWithdrawalData,
+        "orderInfo": withdrawalOutput
+    }
+    closePopup()
+}
```

Comment:
@gschick3 Do you thing here for the withdrawals list (materials tablet) api call